src/Symfony/Component/Routing/Route.php

```diff
@@ -546,6 +546,9 @@ public function compile()
 
     private function sanitizeRequirement(string $key, string $regex)
     {
+        // Replace \A, \z with classic ^ and $
+        $regex = str_replace(['\A', '\z'], ['^', '$'], $regex);
```

Comment:
There is no need to replace all occurrences, only the ones on the edges should be dealt with.
For `\A` it's trivial, but for `\z`, we should deal with escaped backslashes, eg `\\z` should not be replaced.

Reply:
@nicolas-grekas Done. I also removed the type-casting for `substr` result because after checking for empty string `substr($regex, 1)` will always return string.